website/docs/reference/widgets/iframe.md

```diff
@@ -1,100 +1,205 @@
 
-:::info
-When a message is received, Appsmith also executes any code or actions set within the iframe’s [`onMessageReceived` ](iframe.md#events)event. Try it out – set an action within the `onMessageReceived` event, and send another message to watch the results.
-:::
 
-### Events
+### From Appsmith to embedded page
 
-You can define functions that's called when these events are triggered in the widget.
+To send data from Appsmith to an embedded page, you can use the `postMessage()` method in Javascript. This method allows you to send messages between different windows or frames in a web page. When the embedded page receives the message, it can perform actions based on its content. 
 
-| Action                | Description                                                                                                                                                                                                                                                                     |
-| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| **onURLChanged**      | Sets the action to take place when the widget's URL is changed. It can be set from the GUI list of common actions ([examples here](../appsmith-framework/widget-actions/)), or you can define a custom JavaScript function to call instead.                                     |
-| **onSrcDocChanged**   | Sets the action to take place when the `srcDoc`property is changed. It can be set from the GUI list of common actions ([examples here](../appsmith-framework/widget-actions/)), or you can define a custom JavaScript function to call instead. |
-| **onMessageReceived** | Sets the action to take place when a `postMessage` event is received from the embedded page. It can be set from the GUI list of common actions ([examples here](../appsmith-framework/widget-actions/)), or you can define a custom JavaScript function to call instead.        |
+---
+**Example**:  suppose you want to send a message to an embedded page. To achieve this, you need to insert the following code into the **srcDoc** property:
+
+```html
+<div id="target"></div>
+
+<script>
+ window.addEventListener('message', (event) => {
+    const tgt = document.querySelector("#target")
+        tgt.textContent = event.data
+    });
+</script>
+```
+
+This code listens for a message event and updates the text content of an HTML element with the ID `target` based on the data received in the event. When the message event is triggered, the `querySelector()` method is used to find the HTML element, and its `textContent` property is set to the data received in the event.
```

Comment:
can we use the same wording here as this is too verbose and difficult to read- https://appsmith-docs-git-iframe-rehaul-get-appsmith.vercel.app/reference/appsmith-framework/widget-actions/post-message#from-appsmith-to-embedded-page